presto-main/src/main/java/com/facebook/presto/catalogserver/CatalogServer.java

```diff
@@ -147,14 +190,67 @@ public String getMaterializedView(TransactionInfo transactionInfo, SessionRepres
-        if (!referencedMaterializedViewsList.isEmpty()) {
-            return writeValueAsString(referencedMaterializedViewsList, objectMapper);
-        }
-        return EMPTY_STRING;
+        List<QualifiedObjectName> referencedMaterializedViewsList = get(getReferencedMaterializedViewsCache, new CacheKey(session, tableName));
+        return referencedMaterializedViewsList.isEmpty()
+                ? EMPTY_STRING
+                : writeValueAsString(referencedMaterializedViewsList, objectMapper);
+    }
+
+    /*
+        Loading Cache Methods
+     */
+
+    private Boolean loadCatalogExists(CacheKey key)
+    {
+        return metadataProvider.catalogExists(key.getSession().toSession(sessionPropertyManager), (String) key.getKey());
+    }
+
+    private Boolean loadSchemaExists(CacheKey key)
+    {
+        return metadataProvider.schemaExists(key.getSession().toSession(sessionPropertyManager), (CatalogSchemaName) key.getKey());
+    }
+
+    private List<String> loadListSchemaNames(CacheKey key)
+    {
+        return metadataProvider.listSchemaNames(key.getSession().toSession(sessionPropertyManager), (String) key.getKey());
+    }
+
+    private Optional<TableHandle> loadGetTableHandle(CacheKey key)
+    {
+        return metadataProvider.getTableHandle(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private List<QualifiedObjectName> loadListTables(CacheKey key)
+    {
+        return metadataProvider.listTables(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private List<QualifiedObjectName> loadListViews(CacheKey key)
+    {
+        return metadataProvider.listViews(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private Map<QualifiedObjectName, ViewDefinition> loadGetViews(CacheKey key)
+    {
+        return metadataProvider.getViews(key.getSession().toSession(sessionPropertyManager), (QualifiedTablePrefix) key.getKey());
+    }
+
+    private Optional<ViewDefinition> loadGetView(CacheKey key)
+    {
+        return metadataProvider.getView(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private Optional<ConnectorMaterializedViewDefinition> loadGetMaterializedView(CacheKey key)
+    {
+        return metadataProvider.getMaterializedView(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
     }
 
-    private static String writeValueAsString(Object value, ObjectMapper objectMapper)
+    private List<QualifiedObjectName> loadGetReferencedMaterializedViews(CacheKey key)
+    {
+        return metadataProvider.getReferencedMaterializedViews(key.getSession().toSession(sessionPropertyManager), (QualifiedObjectName) key.getKey());
+    }
+
+    private String writeValueAsString(Object value, ObjectMapper objectMapper)
```

Comment:
same comments as the other one. Let's use json or thrift annotation for serde. If it is arbitrary type like `TableHandle`, please use `Any` type. cc: @ajaygeorge

Reply:
I had a discussion with @erick576 a few weeks back regarding this.
TL;DR in the first version they wanted to go with JSON and then refactor it to use `Any`.

See also the discussions on the first PR
https://github.com/prestodb/presto/pull/17855#discussion_r895971940

Reply:
Here is the [issue](https://github.com/prestodb/presto/issues/17928) based on our discussions.